pytest_fixtures/core/sat_cap_factory.py

```diff
@@ -348,13 +349,28 @@ def installer_satellite(request):
     else:
         sat = lru_sat_ready_rhel(getattr(request, 'param', None))
     sat.setup_firewall()
-    # # Register for RHEL8 repos, get Ohsnap repofile, and enable and download satellite
+
+    # register to cdn (also enables rhel repos from cdn)
     sat.register_to_cdn(enable_proxy=True)
-    sat.download_repofile(
-        product='satellite',
-        release=settings.server.version.release,
-        snap=settings.server.version.snap,
-    )
+
+    # setup source repositories
+    if settings.server.version.source == "ga":
+        # enable satellite repos
+        for repo in sat.SATELLITE_CDN_REPOS.values():
+            sat.enable_repo(repo, force=True)
+    else:
+        # get ohsnap repofile
+        sat.download_repofile(
+            product='satellite',
+            release=settings.server.version.release,
+            snap=settings.server.version.snap,
+        )
+    if settings.robottelo.rhel_source == "internal":
+        # disable rhel repos from cdn
```

Comment:
Wouldn't `if elif` be better here instead of trying to disable repos later?

Reply:
`sat.register_to_cdn(enable_proxy=True)` enables AppStream and BaseOS, therefore in case we want to consume Satellite content from CDN, but we want to consume RHEL content from our internal source, we need to disable these two repos.

Reply:
ah ok. I get it.